### ISSUE TITLE
windows sometimes open zoomed (Xorg)

### ISSUE THREAD

**Issue by flokli** [pop-os/shell#1621]
**(1) Issue/Bug Description:**
Windows started opening sometimes as if they were zoomed-in way too much, and don't "fit" into their boundaries they should be using. Probably started happening with GNOME 42, but not entirely sure:

![image](https://user-images.githubusercontent.com/183879/235720824-b4064959-b7df-4ebb-92c7-f039d45f8825.png)

**(2) Steps to reproduce (if you know):**
Hard. Happens 5-6 times per day, and I open a lot of windows :-)

Seems to happen more often with Chrom{e,ium} browsers (or Electron (Slack, vscode)) than terminals, but not Chromium-specific - also had it happen with Telegram.

**(3) Expected behavior:**
Windows to not be scaled 100% and overflowing the space they should be in.

This might be https://github.com/pop-os/shell/issues/1232, reappearing in GNOME 42.

**(4) Distribution (run `cat /etc/os-release`):**
 ```
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
BUILD_ID="23.05pre-git"
DOCUMENTATION_URL="https://nixos.org/learn.html"
HOME_URL="https://nixos.org/"
ID=nixos
LOGO="nix-snowflake"
NAME=NixOS
PRETTY_NAME="NixOS 23.05 (Stoat)"
SUPPORT_URL="https://nixos.org/community.html"
VERSION="23.05 (Stoat)"
VERSION_CODENAME=stoat
VERSION_ID="23.05"
```

**(5) Gnome Shell version:**
GNOME Shell 44.0

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**
pop-shell b5acccefcaa653791d25f70a22c0e04f1858d96e

**(7) Where was Pop Shell installed from:**
nixpkgs

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**
single eDP monitor:

eDP connected primary 1920x1080+0+0 (normal left inverted right x axis y axis) 294mm x 165mm

**(9) Other Installed/Enabled Extensions:**
Removable Drive Menu, Wireless HID

**(10) Other Notes:**
-

**Comment by akumaburn** [comment]
I see something similar happening with IntelliJ Ultimate IDE (may also happen with community but haven't checked) that seems to be reproducible all the time, its consistently overflowing now upon initial project open:

![image](https://github.com/pop-os/shell/assets/4775123/46668e02-8293-4ce5-8cff-16f8332a3fa5)

It's also overflowing left to right:
![image](https://github.com/pop-os/shell/assets/4775123/220b5e07-db88-490d-bc7b-97147aa4e211)

To be clear I'm using the pop-shell extension on AUR ( https://aur.archlinux.org/packages/gnome-shell-extension-pop-shell-git ) so this may be related but not exactly the same issue, thought I'd mention it all the same.

This happens whether or not you select to show the window title bar.
This is with a single display setup.

**Comment by rivenirvana** [comment]
Also happens to Electron apps Caprine and Discord.

**Comment by flokli** [comment]
It might be this only happens when still running with X.Org. I couldn't easily reproduce it on wayland, at least not after 4 attempts. I'll update this issue if I still see it.

**Comment by emko** [comment]
yes same here, all electron apps do this for me, i have to first launch the apps without tiling enabled

**Comment by janith-dev** [comment]
Same here, VS Code and Pycharm CE, all zoomed, mouse is off by couple of mm. as @emko highlighted, if opened without tilling enabled, then it works fine.

**Comment by akumaburn** [comment]
This appears to happen with other tiling extensions like `Forge` as well, so it doesn't appear to be limited to pop-os/shell , I believe that something fundamental in `gnome-shell` has changed (in relation to xorg x11) between gnome 43 and 44. 

I tried looking into it but the documentation for x11 is abysmal.

**Comment by lmassach** [comment]
Same here (Arch, gnome-shell-extension-pop-shell-git from AUR), happens systematically with Quod Libet, Thunderbird and Jellyfin Media Player. If tiling is disabled when I open the window, and then I re-enable tiling once the window is already open, it does ***not*** happen.

**Comment by lutzseverino** [comment]
I'm experiencing this problem now as well in Fedora 38. It's a dumb problem, but it's pretty much a dealbreaker. Zoomed/Stretched programs don't position the mouse correctly and the only workaround is to restart the shell or disable tiling, open the app and enable tiling once again.

**Comment by developomp** [comment]
> I'm experiencing this problem now as well in Fedora 38. It's a dumb problem, but it's pretty much a dealbreaker. Zoomed/Stretched programs don't position the mouse correctly and the only workaround is to restart the shell or disable tiling, open the app and enable tiling once again.

Try maximizing then minimizing the window. That worked for me. Also, I have never had this problem since I've switched from xorg to wayland. It is not a good solution but it works for me.

**Comment by jacobgkau** [comment]
It sounds like nobody experiencing this issue is using Pop!_OS? It would be good to narrow it down to a GNOME Shell or Mutter version above which the issue starts happening, as Pop!_OS 22.04 LTS is still on version 42.5. (Good that it's been narrowed down to Xorg, although Pop!_OS 22.04 LTS still uses Xorg.)

**Comment by italoghost** [comment]
I am having the same issue and it only happens on Xorg. I currently use Arch Linux, with mutter version being 44.1+r2+g82bd40dcbc-1

**Comment by lmassach** [comment]
> It sounds like nobody experiencing this issue is using Pop!_OS? It would be good to narrow it down to a GNOME Shell or Mutter version above which the issue starts happening, as Pop!_OS 22.04 LTS is still on version 42.5. (Good that it's been narrowed down to Xorg, although Pop!_OS 22.04 LTS still uses Xorg.)

I don't think I experienced this before updating from gnome 43 to 44, but I was not really paying attention to the version number

**Comment by matiglesias** [comment]
Same here. I'm running Pop shell on Ubuntu 23 wtih Xorg. The easiest workaround I found is maximizing and unmaximizing the window by hitting Super + Mx2.

Im using NVIDIA 1650 TI on a laptop btw.

**Comment by dbl82** [comment]
I just started seeing this with Pop Shell on Manjaro when I upgraded from GNOME 43 to 44.

**Comment by lutzseverino** [comment]
This issue doesn't seem to be gaining much traction. 

Is this only affecting non-Pop!_OS installations? If so, we might not see a fix in the foreseeable future. 
How about graphic cards? I'm experiencing this issue with an NVIDIA RTX (fedora + nvidia + xorg).

Asking all these questions because this might not be a problem directly caused by shell, further investigation might be in place to direct this issue elsewhere.

**Comment by lmassach** [comment]
I might be mistaken, but it seems to me that it is only affecting GNOME 44, irrespectively of the distro or graphics card (to add to the information I already gave, I am running with Arch + AMD + xorg).

**Comment by mmstick** [comment]
Try disabling animations in the accessibility settings

**Comment by flokli** [comment]
I did that, and it indeed seems to have worked around the issue. Would still make sense to see if this can be fixed in pop shell :-)

**Comment by mmstick** [comment]
Only solution I can offer is the workaround to disable animations when this extension is enabled

**Comment by flokli** [comment]
Is this a known regression to the gnome people? Or something that needs to be handled differently in the extension?

**Comment by mmstick** [comment]
It's not an issue in GNOME 42. Extensions don't have much control over rendering. So I'd say it's a regression in Mutter.

**Comment by emko** [comment]
> It's not an issue in GNOME 42. Extensions don't have much control over rendering. So I'd say it's a regression in Mutter.

i don't think so, try forge, it does not do this

**Comment by akumaburn** [comment]
@mmstick 
Can confirm that checking this Accessibility setting does make the problem go away, albeit at the cost of a jarring user experience.
![image](https://github.com/pop-os/shell/assets/4775123/de56fcb0-68fe-46e8-98cc-3619acdf8e1f)

**Comment by emko** [comment]
i am on fedora 38 not sure what changed but no longer having this issue

**Comment by developomp** [comment]
> i am on fedora 38 not sure what changed but no longer having this issue

What version of mutter and pop shell extension are you on right now?

**Comment by emko** [comment]
> > i am on fedora 38 not sure what changed but no longer having this issue
> 
> What version of mutter and pop shell extension are you on right now?

mutter-44.2-2.fc38.x86_64
gnome-shell-extension-pop-shell-1.2.0^11.b5accce-1.fc38.noarch

**Comment by developomp** [comment]
I assume you're on X11 because it's known that this is not a issue in Wayland.

**Comment by emko** [comment]
> I assume you're on X11 because it's known that this is not a issue in Wayland.

yes i am on X11
![image](https://github.com/pop-os/shell/assets/1070310/a6e86b57-f334-4528-9d32-62e2b1df71f7)

**Comment by akumaburn** [comment]
This seems to have been solved by a recent gnome update, i no longer see this issue even with animations enabled.

Though #1626 is still present.
